core/test/qa_Messages.cpp

```diff
@@ -0,0 +1,286 @@
+#include "gnuradio-4.0/basic/common_blocks.hpp"
+#include <boost/ut.hpp>
+
+#include <gnuradio-4.0/Scheduler.hpp>
+#include <magic_enum.hpp>
+#include <magic_enum_utility.hpp>
```

Comment:
These are some general comments regarding the unit tests:
  * It's good that you refactored and moved the blocks to the blocklib! :+1: 
  * The examples (or blocklib) should include some specific user-level examples how to use the MsgPort in a
    * **fan-in configuration `M->1`**: i.e. multiple `M` producer feeding a single message input port
    * **fan-out configuration `1->N`**: i.e. one message output port of one block connected to `N` message input ports of multiple  other blocks
    * **multicast (fan-in-out) `M -> N`**: where a group  of `M` message output ports/blocks is connected to `N` message input ports/blocks. 
  * brief example how the scheduler implicitly calls the block's process message callback functions as part of the work-loop (assume scheduler is in a 'running' state)
  * brief example how the scheduler could be manually triggered to process messages (assume scheduler is not in a 'running' state).